python/revng/model/migrations/migrate.py

```diff
@@ -0,0 +1,41 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+from revng.model import latest_version  # type: ignore
+
+from revng.model.migrations.versions.v2 import Migrate2
+from revng.model.migrations.versions.v3 import Migrate3
+from .migration import Migration
+
+migrations: dict[int, Migration] = {
+    2: Migrate2(),
+    3: Migrate3(),
+}
+
+
+def migrate(model: dict):
+    if "Version" not in model:
+        return
+
+    version = model["Version"]
+
+    if type(version) is not int:
+        raise ValueError(
+            f".Version field in the model must be of type int but is of type {type(version)}"
+        )
+
+    if version < 0:
+        raise ValueError(f".Version must be a positive integer but is {version}")
+
+    if version > latest_version:
+        raise ValueError(
+            f"The highest possible model schema version is {latest_version}"
+            + f"but the input model has version {version}"
+        )
+
+    if version == 0:
+        return
+
+    for i in range(version + 1, latest_version + 1):
+        migrations[i].migrate(model)
```

Comment:
I'd expect the final version to automatically load the migrations via `importlib.import_module`, without the need for the `migrations` dict

Reply:
Done.